File: javaFiles/Polymorphism.java
package javaFiles;

public class Polymorphism {
    public void carHorn() {
        System.out.println("Beep Beep!");
    }
}

class Car extends Polymorphism {
    public void carHorn() {
        System.out.println("Beep beep! I'm a car!");
    }
}

class Truck extends Polymorphism {
    public void carHorn() {
        System.out.println("Honk Honk!");
    }
}

class Start {
    public static void main(String[] args) {
        Polymorphism myVehicle1 = new Car();
        Polymorphism myVehicle2 = new Truck();

        myVehicle1.carHorn();
        myVehicle2.carHorn();
    }
}

File: dartFiles/Polymorphism.dart
class Horn {
  void carHorn() {
    print("Beep Beep!");
  }
}

class Car extends Horn {
  void carHorn() {
    print("Beep beep! I'm a car!");
  }
}

class Truck extends Horn {
  void carHorn() {
    print("Honk Honk!");
  }
}

void main() {
  Horn myVehicle1 = Car();
  Horn myVehicle2 = Truck();

  myVehicle1.carHorn();
  myVehicle2.carHorn();
}